lib/query/query_supporters.rb

```diff
@@ -506,47 +506,48 @@ def self.dupes_on_email(np_id)
     dupes_expr(np_id)
       .and_where("email IS NOT NULL")
       .and_where("email != ''")
-      .group_by(:email)
+      .group_by("btrim(lower(email, ' '))")
       .execute(format: 'csv')[1..-1]
       .map(&:flatten)
   end
 
   # Find all duplicate supporters by the name column
-  def self.dupes_on_name(np_id)
+  def self.dupes_on_name(np_id, strict_mode = true)
+    group_by_clause = strict_mode ? strict_name_match : loose_name_match
     dupes_expr(np_id)
       .and_where("name IS NOT NULL")
-      .group_by(:name)
+      .group_by(group_by_clause)
       .execute(format: 'csv')[1..-1]
       .map(&:flatten)
   end
 
   # Find all duplicate supporters that match on both name/email
   # @return [Array[Array]] an array containing arrays of the ids of duplicate supporters
-  def self.dupes_on_name_and_email(np_id)
+  def self.dupes_on_name_and_email(np_id, strict_mode = true)
+    group_by_clause = (strict_mode ? strict_name_match : loose_name_match).concat(", btrim(lower(email), ' ')")
     dupes_expr(np_id)
       .and_where("name IS NOT NULL AND email IS NOT NULL AND email != ''")
-      .group_by("name, email")
+      .group_by(group_by_clause)
       .execute(format: 'csv')[1..-1]
       .map(&:flatten)
   end
 
-  def self.dupes_on_name_and_phone(np_id)
+  def self.dupes_on_name_and_phone(np_id, strict_mode = true)
+    group_by_clause = (strict_mode ? strict_name_match : loose_name_match).concat(", phone_index")
```

Comment:
I'm not sure string concatenation is a good idea here. it's so easy to screw up. That said, I probably won't cause any issues.

Reply:
I used it to make sure that we are using the same expressions we wanted everywhere. I started writing the entire expressions and it was so hard to read, that I preferred to do it that way. I don't know if it would make a lot of difference, but we could do this instead:
```suggestion
    group_by_clause = (strict_mode ? strict_name_match : loose_name_match)
    group_by_clause = "#{group_by_clause}, phone_index"
```